src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -2012,28 +2019,26 @@ private static string CheckForMatches(uint[] runningHash, int upTo)
             /// <returns>The string matching the hash, or null.</returns>
             public static string Match(string text)
             {
-                // The longest pattern is 29 characters.
                 // The values in the array are the computed hashes of length
                 // index-1 (so runningHash[0] holds the hash for length 1).
-                var runningHash = new uint[29];
+                Span<uint> runningHash = stackalloc uint[MAX_LEN];
 
                 int longestPossiblePattern = 0;
                 for (int i = 0; i < text.Length; i++)
                 {
+                    // Notice, '-' == ('-' | 0x20)
                     uint h = text[i];
-                    if (h >= 'A' && h <= 'Z')
-                    {
-                        h |= 0x20; // ToLower
-                    }
-                    else if (!((h >= 'a' && h <= 'z') || h == '-'))
+                    h |= 0x20;
+                    if (!((uint)(h - 'a') <= 'z' - 'a' || h == '-'))
```

Comment:
Will blindly doing `h |= 0x20` cause any problem? `h` could be any character, not just ASCII ones.
The original code is way more readable than this, and I double if the perf gain is worth sacrificing the readability.

Reply:
It is not problem because we reject all except A-Z, a-z, and `-`.
Perf win of the change is ~10% because:
- old code optimized for upper case chars
- new code does less operations for any chars.

Reply:
We are talking about improvements in nanosecond level. I don't think it's the bottleneck to the overall script execution, and because of that, I prefer to keep the original code for readability reason.